test/ui/SearchInterfaceTest.ts

```diff
@@ -419,6 +419,51 @@ export function SearchInterfaceTest() {
           expect(simulation.queryBuilder.maximumAge).toBe(123);
         });
       });
+
+      it('allowNoKeywords to true does not cancel the query if there are no keywords', done => {
+        new SearchInterface(div, { allowNoKeywords: true }, undefined, mockWindow);
+        $$(div).on(QueryEvents.doneBuildingQuery, (e, args: IDoneBuildingQueryEventArgs) => {
+          expect(args.cancel).toBe(false);
+          done();
+        });
+        Simulate.query(env);
+      });
+
+      it('allowNoKeywords to false cancels the query if there are no keywords', done => {
+        new SearchInterface(div, { allowNoKeywords: false }, undefined, mockWindow);
+        $$(div).on(QueryEvents.doneBuildingQuery, (e, args: IDoneBuildingQueryEventArgs) => {
+          expect(args.cancel).toBe(true);
+          done();
+        });
+        Simulate.query(env);
+      });
+
+      it('allowNoKeywords to false does not cancel the query if there are keywords', done => {
+        new SearchInterface(div, { allowNoKeywords: false }, undefined, mockWindow);
+        const queryBuilder = new QueryBuilder();
+        queryBuilder.expression.add('foo');
+
+        $$(div).on(QueryEvents.doneBuildingQuery, (e, args: IDoneBuildingQueryEventArgs) => {
+          expect(args.cancel).toBe(false);
+          done();
+        });
+
+        Simulate.query(env, {
+          queryBuilder: queryBuilder
+        });
+      });
+
+      it('allowNoKeywords to false should be sent as a flag in the query', () => {
+        new SearchInterface(div, { allowNoKeywords: false }, undefined, mockWindow);
+        const simulation = Simulate.query(env);
+        expect(simulation.queryBuilder.build().allowNoKeywords).toBe(false);
+      });
+
+      it('allowNoKeywords to true should be sent as a flag in the query', () => {
```

Comment:
I think this test is unnecessary? The name of the test on 456 can be changed to `it('sends allowNoKeywords as a flag in the query')` and then for the expectation:

```typescript
expect(typeof simulation.queryBuilder.build().allowNoKeywords).toBe('Boolean');
```

Reply:
Well, your test could miss the fact that if we set `allowNoKeywords=true` on the search interface, it sets `allowNoKeywords=false` in the query for some reason, and your test would be green, even though it's not doing the right thing.